controllers/volumereplication_controller.go

```diff
@@ -179,6 +179,15 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		return reconcile.Result{}, err
 	}
 
+	// empty `Status.State` conveys that this is the first reconcile, so we enable replication.
+	if instance.Status.State == "" {
+		if err = r.enableReplication(volumeHandle, parameters, secret); err != nil {
```

Comment:
If there is no error let's update the state here. So that it won't go in this if the condition again

Reply:
planning to do that in the status conditions PR - https://github.com/kube-storage/volume-replication-operator/pull/61

Reply:
I'll be updating the conditions and state in the https://github.com/kube-storage/volume-replication-operator/pull/61 PR as well.

Reply:
There is another issue that here that if the `enabling` replication fails, it won't be called again because we are changing the `replicationv1alpha1.State` to `ReplicationFailure`. But that will be fixed in the other PR where we are changing the replication states completely.

Reply:
This should no longer be a problem as we are enabling replication on every reconcile. See [comment](https://github.com/kube-storage/volume-replication-operator/pull/64#discussion_r606921994)